lib/core/src/Cardano/Wallet/Network.hs

```diff
@@ -54,10 +54,13 @@ data NetworkLayer t m = NetworkLayer
         -- ^ Get the current network tip from the chain producer
 
     , postTx
-        :: (Tx t, [TxWitness]) -> ExceptT ErrPostTx m ()
+        :: (tx, [TxWitness]) -> ExceptT ErrPostTx m ()
         -- ^ Broadcast a transaction to the chain producer
     }
 
+instance Functor m => Functor (NetworkLayer m tx) where
+     fmap f nl = nl { nextBlocks = fmap (fmap f) . nextBlocks nl }
```

Comment:
> Restorer will never be a lawful functor though I'm sorry to say

https://github.com/input-output-hk/cardano-wallet/pull/747#issuecomment-534319630

@rvl this is the functor instance I imagined. I just want to map between different block-types, which I think should satisfy `fmap id = id` and `fmap (f . g)  ==  fmap f . fmap g` trivially.

Reply:
OK right very good. So there will be a single separate network layer solely used for maintaining the stake pool metrics?

Reply:
> So there will be a single separate network layer solely used for maintaining the stake pool metrics?

I'd like there to be, yes. As a separation of concerns. Or more of a lightweight separate "view" of the same network layer, since the any potential state (like `MVar UnstableBlocks`) would be the same.

Reply:
> which I think should satisfy fmap id = id and fmap (f . g) == fmap f . fmap g trivially.

I'd say that, as a rule of thumb, if the laws are satisfied (and it's not necessarily obvious), having a small proof in a comment would be the least we could do here :) 
Or, better -> quickcheck properties.

Reply:
Here it is just deriving functor through the composition of `newtype` (the record), `IO` and `ExceptT`. 🤔 

I think the proof would be something unwieldy like
```
-- fmap id nl = nl { nextBlocks = fmap (fmap id) . nextBlocks nl }
-- fmap id nl = nl { nextBlocks = id . nextBlocks nl }
-- fmap id nl = nl { nextBlocks = nextBlocks nl }
-- fmap id nl = nl
--
-- fmap (f . g) nl = nl { nextBlocks = fmap (fmap (f . g)) . nextBlocks nl }
-- fmap (f . g) nl = nl { nextBlocks = fmap ((fmap f) . (fmap g)) . nextBlocks nl }
-- fmap (f . g) nl = nl { nextBlocks = (fmap (fmap f) . fmap (fmap g)) . nextBlocks nl }
-- fmap (f . g) nl = 
--      let nl' { nextBlocks =  fmap (fmap g)) . nextBlocks nl }
--      in nl' { nextBlocks = fmap (fmap f) . nextBlocks nl }
```